ledger.go

```diff
@@ -692,459 +736,486 @@ FINALIZE_ROUNDS:
 }
 
 func (l *Ledger) SyncToLatestRound() {
+	defer l.stopWG.Done()
+
 	voteWG := new(sync.WaitGroup)
 
+	shutdown := func() {
+		close(l.sync)
+		l.consensus.Wait() // Wait for all consensus-related workers to shutdown.
+
+		voteWG.Add(1)
+		close(l.syncVotes)
+		voteWG.Wait() // Wait for the vote processor worker to shutdown.
+
+		l.finalizer.Reset() // Reset consensus Snowball sampler.
+		l.syncer.Reset()    // Reset syncing Snowball sampler.
+	}
+
+	restart := func() { // Respawn all previously stopped workers.
+		l.syncVotes = make(chan vote, sys.SnowballK)
+		go CollectVotes(l.accounts, l.syncer, l.syncVotes, voteWG)
+
+		l.sync = make(chan struct{})
+		l.PerformConsensus()
+	}
+
 	go CollectVotes(l.accounts, l.syncer, l.syncVotes, voteWG)
 
 	for {
-		for {
-			conns, err := SelectPeers(l.client.ClosestPeers(), sys.SnowballK)
-			if err != nil {
+		select {
+		case <-l.stop:
+			shutdown()
+			return
+
+		default:
```

Comment:
maybe leave default branch empty, so code will fall through and there will be no need in additional nesting level?

Reply:
Yeah that would be much better.